controllers/front/callback.php

```diff
@@ -63,30 +64,59 @@ public function postProcess()
             $this->jsonResponse('Payment module unavailable', true, 403);
         }
 
-        $total = (float)$cart->getOrderTotal(true, Cart::BOTH);
-        $currency = $this->context->currency;
+        // Get orderId
+        $orderId = OrderCore::getOrderByCartId($cart->id);
+
+        // Check if order creation is already happening
+        if ($orderId === false && $onpay->isCartLocked($cart->id)) {
+            // Wait for order creation to end for 500ms, and try to get order again.
```

Comment:
You write 500ms, but you actually do 500µs ?

Reply:
That's an error. I've tweaked this sleep time to 1s, since TB can be a bit slow.